### PR TITLE
Unsaferow size calculator

### PR DESCRIPTION
**Pull request by beroyfb**

### CODE REVIEW

File: velox/row/UnsafeRowDynamicSerializer.h [hunk 1]
```diff
@@ -288,6 +288,218 @@ struct UnsafeRowDynamicSerializer : UnsafeRowSerializer {
...
+  /// \param data The elements vector
+  /// \return
+  template <TypeKind Kind>
+  inline static size_t getSizeSimpleVector(
+      size_t size,
+      const BaseVector* data) {
+    using NativeType = typename TypeTraits<Kind>::NativeType;
+    const auto& simple =
+        *data->loadedVector()->asUnchecked<SimpleVector<NativeType>>();
+    size_t dataSize = size * sizeof(NativeType);
```
[the remark this answers is not in the document]
Comment (beroyfb) [reply velox/row/UnsafeRowDynamicSerializer.h:309]: Good question. It's taken care of inside ```getSizeArray``` which calls this:
```
    result +=
        nullLength + getSizeVector(elementsType, offset, size, elementsVector);

```

File: velox/row/UnsafeRowDynamicSerializer.h [hunk 2]
```diff
@@ -288,6 +288,218 @@ struct UnsafeRowDynamicSerializer : UnsafeRowSerializer {
...
+  }
+
+  /// Gets the size of a range of elements in an array
+  /// \param elementsType type of elements
+  /// \param offset the starting offset of elements
+  /// \param size the size of elements
+  /// \param elementsVector element vector
+  /// \return
+  inline static size_t getSizeArray(
+      const TypePtr& elementsType,
```
[the remark this answers is not in the document]
Comment (beroyfb) [reply velox/row/UnsafeRowDynamicSerializer.h:335]: It's a dynamic serializer. How do you suggest to change it exactly?

File: velox/row/UnsafeRowDynamicSerializer.h [hunk 3]
```diff
@@ -288,6 +288,218 @@ struct UnsafeRowDynamicSerializer : UnsafeRowSerializer {
...
+    }
+  }
+
+  /// Gets the size of an element in a vector
+  /// \param type the of the element
+  /// \param data the vector
+  /// \param idx the index of the element in the vector
+  /// \return
+  inline static size_t
+  getSizeArray(const TypePtr& type, const ArrayVector* data, size_t idx) {
```
[the remark this answers is not in the document]
Comment (beroyfb) [reply velox/row/UnsafeRowDynamicSerializer.h:397]: It's only used when wrappedVector called is passed to it which comes as * (not Ptr). I can create a shared Ptr out of it if you think it's necessary.

File: velox/row/UnsafeRowDynamicSerializer.h [hunk 4]
```diff
@@ -288,6 +288,218 @@ struct UnsafeRowDynamicSerializer : UnsafeRowSerializer {
...
+  /// \param data the vector containing the element
+  /// \param idx the index of the element in the vector
+  /// \return
+  inline static size_t getSizeStringView(const VectorPtr& data, size_t idx) {
+    auto* rawData = data->loadedVector();
+    const auto& simple = static_cast<const SimpleVector<StringView>&>(*rawData);
+    if (simple.isNullAt(idx)) {
+      return 0;
+    }
+    auto stringSize = simple.valueAt(idx).size();
```
[the remark this answers is not in the document]
Comment (kgpai) [reply velox/row/UnsafeRowDynamicSerializer.h:323]: The cast itself should be a compile time thing, but maybe we can gate the call to loaded based on some boolean - I believe there is, isloaded or something equivalent ?

**Review by kgpai — APPROVED** [review]

**Pull request closed without merging**